src/Features/CSharp/Portable/CodeFixes/HideBase/HideBaseCodeFixProvider.AddNewKeywordAction.cs

```diff
@@ -25,16 +33,35 @@ protected override async Task<Document> GetChangedDocumentAsync(CancellationToke
             {
                 var root = await _document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
-                var newNode = GetNewNode(_node);
+                var newNode = await GetNewNodeAsync(_node, cancellationToken).ConfigureAwait(false);
                 var newRoot = root.ReplaceNode(_node, newNode);
 
                 return _document.WithSyntaxRoot(newRoot);
             }
 
-            private SyntaxNode GetNewNode(SyntaxNode node)
+            private async Task<SyntaxNode> GetNewNodeAsync(SyntaxNode node, CancellationToken cancellationToken)
             {
-                var generator = SyntaxGenerator.GetGenerator(_document);
-                return generator.WithModifiers(node, generator.GetModifiers(node).WithIsNew(true));
+                var modifiers = SyntaxFacts.GetModifiers(node);
+                var newModifiers = modifiers.Add(SyntaxFactory.Token(SyntaxKind.NewKeyword));
+
+                var options = await _document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+                var option = options.GetOption(PreferredModifierOrderOption);
+                if (!OrderModifiersHelper.TryGetOrComputePreferredOrder(option.Value, out var preferredOrder) ||
+                    !CodeAnalysis.OrderModifiers.AbstractOrderModifiersHelpers.IsOrdered(preferredOrder, modifiers))
+                {
+                    return SyntaxFacts.WithModifiers(node, newModifiers);
```

Comment:
💭 It seems like this should use the `SyntaxGenerator` approach from the original code. Do you see any reason why we wouldn't do that?

Reply:
`SyntaxGenerator.WithModifiers` will "reorder" the existing modifiers. Is this expected?

Reply:
`static public` -> `public new static`